### ISSUE TITLE
Renew command does not only update "about to expire" certificates.

### ISSUE THREAD

**Issue by SamMousa** [certbot/certbot#2531]
As far as I understand `./letsencrypt-auto renew` should try to renew only certificates that are about to expire.
When during the renew succeeds for some domains, but fails for others, re-running it will also try to renew the certificates that already have been successfully renewed.

Basically when running the command X times there should never be a situation where rate limit error is thrown for a certificate that has already been renewed.
Example, my certificate at `https://bugtracker.befound.nl` was renewed, but when I (re)run the renew command I get this:

```
2016-02-23 10:55:45,925:WARNING:letsencrypt.cli:Attempting to renew cert from /etc/letsencrypt/renewal/bugtracker.befound.nl.conf produced an unexpected error: urn:acme:error:rateLimited :: There were too many requests of a given type :: Error creating new cert :: Too many certificates already issued for: befound.nl. Skipping.
```

(I have verified in `/etc/letsencrypt/renewal` that there is only 1 file containing that domain name)

**Comment by pde** [comment]
Currently it's configured to renew 30 days before expiry, which gives time to investigate if something goes wrong, and prevents errors from clients whose clock times are several weeks out (which turns out to be quite a lot of clients).  Though it isn't *officially* supported yet, you can put

`renew_before_expiry = 10 days ` (or whatever you want) in the top section of `/etc/letsencrypt/renewal/bugtracker.befound.nl.conf` to change this

**Comment by pde** [comment]
If you had <30 days left on your cert, I think this is a wontfix.  If you had >30 days on your cert, did you use `--renew-by-default`?

**Comment by SamMousa** [comment]
@pde Not sure about `--renew-by-default`, can't find that mentioned in the docs. 
What I did was the following:
1. I got an email saying certificate A and B are about to expire.
2. I run `./letsencrypt-auto renew` 
3. It renews A but throws an error on B.
4. I make some configuration changes in `/etc/letsencrypt/renewal/B`
5. I run `./letsencrypt-auto renew` 

That is when weird stuff happens. Certificate A was already renewed so step 5 should only try to renew B, but instead it tries to renew A as well, causing me to hit a rate limit if I attempt this several times...
Seems to me that the renewal / expiry information is not updated per renewed certificate but only when all renewals succeed.
